pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1152,122 +1235,168 @@ func (lbaas *LbaasV2) ensureSecurityGroup(clusterName string, apiService *v1.Ser
 		}
 		lbSecGroupID = lbSecGroup.ID
 
-		//add rule in security group
-		for _, port := range ports {
-			for _, sourceRange := range sourceRanges.StringSlice() {
-				ethertype := rules.EtherType4
-				network, _, err := net.ParseCIDR(sourceRange)
+		// If Octavia is used, the VIP port security group is already taken good care of, we only need to allow ingress
+		// traffic from Octavia amphorae to the node port on the worker nodes.
+		if !lbaas.opts.UseOctavia {
```

Comment:
if octavia is used but this ! means that octavia is not used?

Reply:
The comment is just for describing the checking purpose, i will change a little bit.